### DISCUSSION TITLE
Q: How to set friendly name of a device from a client?

### DISCUSSION THREAD

**Discussion by jvcdk** [Koenkk/zigbee2mqtt#22907] [General]
Hi all,

I am writing my own little light management software; I am using Zigbee devices (buttons and lights) and I am using ZigbeeToMqtt to communicate with the Zigbee coordinator / devices.

I would like to be able to change the friendly name of my devices. I see that there is a configuration file, but I would like not to edit it. What I would like is to change the name via my own written application / UI.

I have tried various things, like:
* Publishing to `zigbee2mqtt/<device address>/set` with payload `{ "friendly_name": "<new name>" }`
* Publishing to `zigbee2mqtt/<device address>/set/friendly_name` with payload `"<new name>"`
* Publishing to `zigbee2mqtt/<device address>` with payload `{ "friendly_name": "<new name>" }`
* Variations of above with capitalizing friendly_name, using "name", and "Name" etc.

Nothing seems to work. For the two first lines, I get a response from Z2M: `No converter available for 'friendly_name' ("<new name>")`.

What is the correct way (if possible) to configure a device friendly name over the bus?

BR
Jørn

**Comment by lux73** [comment]
~~AFAIK not possible atm~~

/edit
oh, there is a way - notice the Post form @francisp2 😎

**Comment by francisp2** [comment]
zigbee2mqtt/bridge/request/device/rename

see 

https://www.zigbee2mqtt.io/guide/usage/mqtt_topics_and_messages.html#zigbee2mqtt-bridge-request-device-rename

**Reply by jvcdk** [reply]
Amazing. Thank you :) I have been looking at that page; don't know why I missed it.
